### PR TITLE
HIVE-24530: Potential NPE in FileSinkOperator.closeRecordwriters method

### PR DESCRIPTION
**Pull request by kuczoram**

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/exec/FileSinkOperator.java [hunk 1]
```diff
@@ -1204,8 +1205,10 @@ private void closeWriters(boolean abort) throws HiveException {
   private void closeRecordwriters(boolean abort) {
     for (RecordWriter writer : rowOutWriters) {
```
Comment (szlta) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/FileSinkOperator.java:1206]: So now there is a null check for the RecordWriter instances in this list, but shouldn't we also check for the list itself (rowOutWriters) being null here?

**Review by szlta — COMMENTED** [review]
One minor comment, otherwise looking good!

File: ql/src/java/org/apache/hadoop/hive/ql/exec/FileSinkOperator.java (hunk 1 above)
Comment (kuczoram) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/FileSinkOperator.java]: You're right, I added a null check for the rowOutWriters as well.

**Review by szlta — APPROVED** [review]
+1 pending tests

**Pull request merged**